### DISCUSSION TITLE
Http-client TaskLet not evaling properties

### DISCUSSION THREAD

**Discussion by gaberger** [mikub/titanoboa#21] [General]
I am using the docker container which is at 0.9.0. clj-http requires `basic-auth` key for authorization. 

This is not evaling

`:basic-auth2 #titanoboa.exp/Expression{:value " [(:FWD_USER *properties*) (:API_TOKEN *properties*)] ",
                                                              :type "clojure"},`

**Comment by mikub** [comment]
Hi @gaberger, thanks for letting me know.
I ran a quick test on the hosted instance which is at 0.9.0 as well and it seems to be working. Could you maybe post the whole workflow definition edn?

Mine is:
```clojure
{:first-step "step",
 :properties {:API_TOKEN "token", :FWD_USER "user"},
 :revision 2,
 :name "demo3",
 :steps [{:properties {:as :json,
                       :url #titanoboa.exp/Expression{:value "(mapv #(str \"https://jsonplaceholder.typicode.com/posts/\"  %) (range 1 10))",
                                                      :type nil},
                       :body-only? true,
                       :response-property-name :rest-response,
                       :basic-auth2 #titanoboa.exp/Expression{:value " [(:FWD_USER *properties*) (:API_TOKEN *properties*)] ",
                                                              :type "clojure"},
                       :request-method :get},
          :revision 1,
          :type :http-client,
          :next [],
          :id "step",
          :supertype :tasklet,
          :workload-fn #titanoboa.exp/Expression{:value "io.titanoboa.tasklet.httpclient/request",
                                                 :type "clojure"}}],
 :type nil}
```

I can see that the :basic-auth2 get resolved to the vector with no issues.

My ext dependencies are:
```clojure
{:coordinates [[io.titanoboa.tasklet/http-client "0.1.1"]],
 :require [[io.titanoboa.tasklet.httpclient]],
 :repositories {"clojars" "https://clojars.org/repo",
                "central" "https://repo1.maven.org/maven2/"},
 :import [[nil]]}
```

The question might be where you defined `:API_TOKEN` and `:FWD_USER` - whether in the same step you defined also `:basic-auth2`?
Also another question is whether the `clj-http.client/request` picks that up once it gets resolved, but I assume we are not that far yet.

Cheers
Miro
